server/src/main/java/org/tctalent/server/service/db/impl/UserServiceImpl.java

```diff
@@ -610,6 +615,14 @@ public void updatePassword(UpdateUserPasswordRequest request) {
         String passwordEnc = passwordHelper.validateAndEncodePassword(request.getPassword());
         user.setPasswordEnc(passwordEnc);
         userRepository.save(user);
+
+        /* Update changePassword */
+        if (user.getCandidate().isChangePassword()) {
+            Candidate candidate = candidateRepository.findById(user.getCandidate().getId())
```

Comment:
This will throw a null pointer exception for users that aren't candidates.

if (user.getCandidate() != null &&

Reply:
done